whitepapers/tickets/SHARE-822.md

```diff
@@ -0,0 +1,113 @@
+| date_modified | datetime |  Desc   |          |     |         |                                                                                                    |
+
+| id  | name       | deleted | handle    | taxonomy_id | bepress_id | parent_id |
+| --- | ---------- | ------- | --------- | ----------- | ---------- | --------- |
+| 1   | Law        | False   | Law       | 1           | 1          | NULL      |
+| 2   | Civil Law  | False   | Civil Law | 1           | 2          | 1         |
+| 3   | Custom Law | False   | 538df3e1  | 2           | 1          | 1         |
+
+## Changes to OSF: 
+  - Whenever a custom taxonomy if modified the ne
+  - The on_preprint updated method will now serialize subjects as:
+
+## Situations:
+
+### Reference
+```
+Taxonomy X:
+  Subject Alpha -> Bepress A
+    Subject Bravo -> Bepress B
+      Subject Charlie -> Bepress C
+    Subject Delta -> Bepress B
+  Subject Echo -> Bepress D
+
+
+Given SHARE is aware of custom Taxonomy X
+When SHARE recieves a work, "Gamma", with subjects [Alpha, Bravo, Charlie]
+Then SHARE will remove all parent subjects as they will be implied by subject "Charlie"
+
+  Reasoning:
+    If at any point "Charlie" is moved to another location in Taxonomy X no changes need to
+    be made to "Gamma", we simply find all works that haven't been modified since "Charlie"
+    has been modified and re-index them.
+
+Given SHARE is aware of custom Taxonomy X
+When SHARE recieves a work, "Gamma", with subjects [Charlie, Bepress C, Bepress D]
+Then SHARE will not modify the submitted subjects
+
+  Reasoning:
+    "Charlie" may currently imply that "Bepress C" is attached to "Gamma". If "Charlie" is
+    ever redefined to be a different Bepress subject that implication will no longer be true.
+
+  Ramifications:
+    If "Gamma" is being submitted with "Bepress C" because "Charlie" implies it, "Bepress C" will
+    have to be explicitly removed if "Charlie" is redefined.
+    Works should only be submitted with subjects explicitly assigned by the user.
+
+Given SHARE is aware of custom Taxonomy X and "Gamma" exists with subjects [Charlie, Delta]
+When SHARE is notified that "Delta" has been deleted
+Then SHARE will mark Delta as deleted and reindex all works attached to "Delta"
+
+  Reasoning:
+    "Delta" will have to exist for SHARE to detect the need to remove the subject from existing
+    works.
+ ```
+### Strange Situations:
+  - Custom Taxonomy item added
+    - Foxtrot is added to Taxonomy X
+  - Custom Taxonomy item removed
+    - Bravo is removed to Taxonomy X
+    - Echo is removed to Taxonomy X
+  - Custom Taxonomy item renamed
+    - Echo -> Romeo is added to Taxonomy X
+  - Custom Taxonomy item redefined
+    - Alpha -> Bepress B is added to Taxonomy X
+  - Custom Taxonomy item duplicated
+    - Foxtrot -> Bepress A is added to Taxonomy X
+  - Preprint with Custom Taxonomy moves providers
```

Comment:
I think this would be another situation where it would be updated on the source.